# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/TypedArrayObject.cpp
# Commit: 89d2b65aaae2
# Full Hash: 89d2b65aaae2baad569c98cb676f305539365ef8
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-11-30 21:44:22
# Description:
#   Bug 1743441 - Missing nullptr check in TypedArray_base::fromObject r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132416
# ==============================================================================

diff -r 6ff1d34c32c6 -r 89d2b65aaae2 js/src/vm/TypedArrayObject.cpp
--- a/js/src/vm/TypedArrayObject.cpp	Tue Nov 30 15:43:36 2021 +0000
+++ b/js/src/vm/TypedArrayObject.cpp	Tue Nov 30 15:51:27 2021 +0000
@@ -2819,7 +2819,7 @@
 }
 
 JS::TypedArray_base JS::TypedArray_base::fromObject(JSObject* unwrapped) {
-  if (unwrapped->is<TypedArrayObject>()) {
+  if (unwrapped && unwrapped->is<TypedArrayObject>()) {
     return TypedArray_base(unwrapped);
   }
   return TypedArray_base(nullptr);
